backend/controllers/productController.js

```
@@ -147,18 +129,40 @@ const singleProduct = asyncHandler(async (req, res) => {
-    const filter = { name: { $regex: search, $options: "i" } };
+    // Apply category filter if selected
+    if (category) {
+      filter.category = { $in: category.split(",") };
+    }
+
+    // Apply subcategory filter if selected
+    if (subCategory) {
+      filter.subCategory = { $in: subCategory.split(",") };
+    }
+
+    let sortOption = {};
```

Biome (1.9.4)

[error] This let declares a variable that is only assigned once.

'sortOption' is never reassigned.

Safe fix: Use const instead.


(lint/style/useConst)